lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlobFileSystem.cs

```diff
@@ -0,0 +1,255 @@
+            {
+                throw new ArgumentOutOfRangeException(@"URI must be a file and cannot be a container.");
+            }
+
+            try
+            {
+                _client.GetBlobReferenceFromServer(fileUri).Delete();
+            }
+            catch (StorageException e)
+            {
+                if (e.RequestInformation.HttpStatusCode != (int) HttpStatusCode.NotFound)
+                {
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// <see cref="IFileSystem.Exists"/>
+        /// </summary>
+        public bool Exists(Uri fileUri)
+        {
+            try
+            {
+                return _client.GetBlobReferenceFromServer(fileUri).Exists();
+            }
+            catch (StorageException e)
+            {
+                if (e.RequestInformation.HttpStatusCode != (int) HttpStatusCode.NotFound)
+                {
+                    throw;
+                }
+
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Copies to a CloudBlockBlob from another block blob.
+        /// <see cref="IFileSystem.Copy"/>
+        /// </summary>
+        public void Copy(Uri sourceUri, Uri destinationUri)
+        {
+            new CloudBlockBlob(destinationUri, _client.Credentials).StartCopyFromBlobAsync(sourceUri).GetAwaiter().GetResult();
+        }
+
+        /// <summary>
+        /// <see cref="IFileSystem.CopyToLocal"/>
+        /// </summary>
+        public void CopyToLocal(Uri remoteFileUri, string localName)
+        {
+            _client.GetBlobReferenceFromServer(remoteFileUri).DownloadToFile(localName, FileMode.CreateNew);
+        }
+
+        /// <summary>
+        // Copies to a CloudBlockBlob from a local file.
+        /// <see cref="IFileSystem.CopyFromLocal"/>
+        /// </summary>
+        public void CopyFromLocal(string localFileName, Uri remoteFileUri)
+        {
+            new CloudBlockBlob(remoteFileUri, _client.Credentials).UploadFromFile(localFileName, FileMode.Open);
+        }
+
+        /// <summary>
+        /// Azure blobs does not have the sense of the existence of a "directory."
+        /// We will not throw an error, but we will not do anything either.
+        /// </summary>
+        public void CreateDirectory(Uri directoryUri)
+        {
+        }
+
+        /// <summary>
+        /// Recursively deletes blobs under a specified "directory URI."
+        /// If only the container is specified, the entire container is deleted.
+        /// </summary>
+        public void DeleteDirectory(Uri directoryUri)
+        {
+            var uriSplit = directoryUri.AbsolutePath.Split(new[] {"/"}, StringSplitOptions.RemoveEmptyEntries);
+            if (!uriSplit.Any())
+            {
+                throw new ArgumentOutOfRangeException(@"URI must contain at least the container");
+            }
+
+            var containerName = uriSplit[0];
```

Comment:
No, you want the container name here, not the host.
e.g. for http://host.com/container/blob, you want "container" here not "host.com"